server/etcdserver/version/monitor.go

```diff
@@ -0,0 +1,141 @@
+package version
+
+import (
+	"github.com/coreos/go-semver/semver"
+	"go.etcd.io/etcd/api/v3/version"
+	"go.etcd.io/etcd/server/v3/etcdserver/api/membership"
+	"go.uber.org/zap"
+)
+
+type VersionMonitor struct {
```

Comment:
docs on public struct and interface?

Reply:
Also the package name is already `version`. How about `Monitor`? And `Storage` for `VersionStorage`?

Reply:
Done